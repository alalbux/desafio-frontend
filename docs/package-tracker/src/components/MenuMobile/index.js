'use client';
import React from 'react';
import Drawer from '@mui/material/Drawer';
import List from '@mui/material/List';
import ListItem from '@mui/material/ListItem';
import Link from '@mui/material/Link';
import LOCALES from '@/locales/navigation';

const { MENU } = LOCALES;

const MenuMobile = ({ isDrawerOpen, toggleDrawer, isOpen }) => {
  return (
    <Drawer
      anchor="top"
      open={isDrawerOpen && isOpen}
      onClose={toggleDrawer(false)}
    >
      <List>
        {MENU.map((item, index) => (
          <ListItem key={item.text} onClick={toggleDrawer(false)}>
            <Link variant="button" color="text.primary" href={item.description}>
              {item.text}
            </Link>
          </ListItem>
        ))}
      </List>
    </Drawer>
  );
};

export default MenuMobile;
