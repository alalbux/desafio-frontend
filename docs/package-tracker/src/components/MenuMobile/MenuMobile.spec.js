import React from 'react';
import { render } from '@testing-library/react';
import MenuMobile from '@/components/MenuMobile';

describe('MenuMobile', () => {
  it('matches the snapshot', () => {
    const toggleDrawer = jest.fn();
    const { container } = render(<MenuMobile toggleDrawer={toggleDrawer} />);
    expect(container).toMatchSnapshot();
  });
});
