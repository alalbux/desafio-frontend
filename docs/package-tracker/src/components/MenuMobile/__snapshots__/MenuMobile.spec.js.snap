// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuMobile matches the snapshot 1`] = `<div />`;
